rfc/src/rfcs/0008-viewerless-coverage.md

```diff
@@ -0,0 +1,92 @@
+
+We anticipate missing locations being the main corner case for the coverage feature.
+However, coverage reported through `cbmc-viewer` also suffers from this issue.
+This will be mitigated by ignoring checks coming from unknown locations and propagating locations if possible whenever they're missing (as suggested in [this comment](https://github.com/model-checking/kani/issues/2048#issuecomment-1599680694)).
+
+## Rationale and alternatives
+
+The main advantage of this design is that it'll give Kani full control over its coverage-related options.
+This implies that not only it'll be able to select the mechanism to collect coverage information, but also have different options to output coverage data.
+All of that fully integrated in Kani and tested through `compiletest`.
+In contrast, this means that Kani will become more complex, so the development and maintainance burden will be higher.
+
+This feature is expected to have a substantial impact for Kani.
+Users frequently ask for coverage-related options but Kani doesn't have one.
+That's because, some months ago, we noticed incorrect coverage results were being reported with the `--visualize` option, prompting us to disable the coverage reports emitted by `cbmc-viewer`.
+
+## Open questions
+
+Open questions:
+ * Do we want a `--coverage` option or a `coverage` subcommand? Coverage can be seen as a different workflow but also a verification option.
+
+## Future possibilities
+
+We expect many incremental improvements in the coverage area in subsequent versions:
+ 1. Replacing the injection mechanism proposed in this RFC with the Rust compiler APIs for coverage (e.g., [CodeRegion](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/coverage/struct.CodeRegion.html) and/or a MIR pass similar to [InstrumentCoverage](https://doc.rust-lang.org/stable/nightly-rustc/rustc_mir_transform/coverage/struct.InstrumentCoverage.html)) so we can retrieve region-based coverage information. Note that this is a requirement for items (2) and (3) below.
+ 2. Displaying region-based coverage information similar to [Rust's instrument-based code coverage](https://doc.rust-lang.org/rustc/instrument-coverage.html), which allows users to see coverage information for more granular parts of code (e.g., subconditions hit in a disjunctive condition).
+ 3. Adding new user-requested coverage formats such as GCOV [#1706](https://github.com/model-checking/kani/issues/1706) or LCOV [#1777](https://github.com/model-checking/kani/issues/1777).
+ 4. Enabling an option to run verification and coverage at the same time, so users can obtain both results at the same time.
+ 5. Performing optimization improvements to `kani-compiler` and its engines to speed up the data collected for coverage information.
+
+
+[^coverage-assertions]: Currently, we replace non-coverage assertions with assumptions so the execution is blocked instead of reporting a failure.
```

Comment:
This should be part of the design section. If it's not clear whether this is something we want, please also add it to the open questions.